packages/compiler/src/core/name-resolver.ts

```diff
@@ -0,0 +1,1306 @@
+
+      if (~sourceSym.flags & SymbolFlags.Enum) {
+        // will be a checker error
+        continue;
+      }
+
+      const sourceTable = getAugmentedSymbolTable(sourceSym.members!);
+      targetTable.include(sourceTable, enumSym);
+    }
+  }
+
+  function bindUnionMembers(node: UnionStatementNode) {
+    // Everything is already bound in binder.ts
+  }
+  function bindScalarMembers(node: ScalarStatementNode) {
+    const scalarSym = node.symbol!;
+    const targetTable = getAugmentedSymbolTable(scalarSym.members!);
+    const scalarSymLinks = getSymbolLinks(scalarSym);
+
+    if (node.extends) {
+      const { finalSymbol: extendsSym, resolutionResult: extendsResult } = resolveTypeReference(
+        node.extends,
+      );
+      setUnknownMembers(scalarSymLinks, extendsSym, extendsResult);
+
+      if (~extendsResult & ResolutionResultFlags.Resolved) {
+        return;
+      }
+      compilerAssert(extendsSym, "Scalar extends symbol must be defined if resolution succeeded");
+
+      const sourceTable = getAugmentedSymbolTable(extendsSym.members!);
+      targetTable.include(sourceTable, scalarSym);
+    }
+  }
+
+  function bindTemplateParameter(node: TemplateParameterDeclarationNode) {
+    const sym = node.symbol;
+    const links = getSymbolLinks(sym);
+    links.hasUnknownMembers = true;
+  }
+
+  function resolveIdentifier(
+    node: IdentifierNode,
+    options: ResolveTypReferenceOptions,
+  ): ResolutionResult {
+    let scope: Node | undefined = node.parent;
+    let binding: Sym | undefined;
+
+    while (scope && scope.kind !== SyntaxKind.TypeSpecScript) {
+      if (scope.symbol && scope.symbol.flags & SymbolFlags.ExportContainer) {
+        const mergedSymbol = getMergedSymbol(scope.symbol);
+        binding = tableLookup(mergedSymbol.exports!, node, options.resolveDecorators);
+        if (binding) return resolvedResult(binding);
+      }
+
+      if ("locals" in scope && scope.locals !== undefined) {
+        binding = tableLookup(scope.locals, node, options.resolveDecorators);
+        if (binding) {
+          return resolvedResult(binding);
+        }
+      }
+
+      scope = scope.parent;
+    }
+
+    if (!binding && scope && scope.kind === SyntaxKind.TypeSpecScript) {
+      // check any blockless namespace decls
+      for (const ns of scope.inScopeNamespaces) {
+        const mergedSymbol = getMergedSymbol(ns.symbol);
+        binding = tableLookup(mergedSymbol.exports!, node, options.resolveDecorators);
+
+        if (binding) return resolvedResult(binding);
+      }
+
+      // check "global scope" declarations
+      const globalBinding = tableLookup(
+        globalNamespaceNode.symbol.exports!,
+        node,
+        options.resolveDecorators,
+      );
+
+      // check using types
+      const usingBinding = tableLookup(scope.locals, node, options.resolveDecorators);
+
+      if (globalBinding && usingBinding) {
+        reportAmbiguousIdentifier(node, [globalBinding, usingBinding]);
```

Comment:
Need to remove those, should be able to add to symol links the list of ambigous symbols?